lib/db.c

```diff
@@ -74,12 +74,79 @@ static const uint32_t GRN_TABLE_PAT_KEY_CACHE_SIZE = 1 << 15;
       grn_string_get_normalized(ctx, nstr, &key, &key_size, NULL);\
       block\
       grn_obj_close(ctx, nstr);\
+    } else {\
+      GRN_DEFINE_NAME(table);\
+      ERR(GRN_INVALID_ARGUMENT,\
+          "[key][normalize] failed to normalize: <%.*s> <%.*s>",\
+          (int)name_size, (const char *)name,\
+          (int)key_size, (const char *)key); \
     }\
   } else {\
     block\
   }\
 } while (0)
 
+#define INSPECT_TABLE_LIMITED_WITH_KEY(message, table, key, key_size) do { \
```

Comment:
There is an extra space between `;` and `\`.

Reply:
We should not use large macro  because it's difficult to debug.
Why don't you use function?

Reply:
I've fixed to use function.